### ISSUE TITLE
weak_delegate violation does not work

### ISSUE THREAD

**Issue by vladyslavsosiuk** [realm/SwiftLint#4079]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

When you declare a delegate property without weak keyword swiftlint does not produce warning

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint       
Linting Swift files in current working directory
Linting 'ViewController.swift' (1/3)
Linting 'AppDelegate.swift' (2/3)
Linting 'SceneDelegate.swift' (3/3)
/XcodeProjects/Prototypes/SwiftlintWeakDelegateIssue/SwiftlintWeakDelegateIssue/ViewController.swift:17:5: warning: Identifier Name Violation: Variable name should be between 3 and 40 characters long: 'propertyWithUnusedSetterToSeeIfSwiftlintWorks' (identifier_name)
/XcodeProjects/Prototypes/SwiftlintWeakDelegateIssue/SwiftlintWeakDelegateIssue/ViewController.swift:19:9: warning: Unused Setter Value Violation: Setter value is not used. (unused_setter_value)
/XcodeProjects/Prototypes/SwiftlintWeakDelegateIssue/SwiftlintWeakDelegateIssue/AppDelegate.swift:15:1: warning: Line Length Violation: Line should be 120 characters or less: currently 145 characters (line_length)
/XcodeProjects/Prototypes/SwiftlintWeakDelegateIssue/SwiftlintWeakDelegateIssue/AppDelegate.swift:22:1: warning: Line Length Violation: Line should be 120 characters or less: currently 179 characters (line_length)
/XcodeProjects/Prototypes/SwiftlintWeakDelegateIssue/SwiftlintWeakDelegateIssue/AppDelegate.swift:30:1: warning: Line Length Violation: Line should be 120 characters or less: currently 157 characters (line_length)
/XcodeProjects/Prototypes/SwiftlintWeakDelegateIssue/SwiftlintWeakDelegateIssue/AppDelegate.swift:36:1: warning: Trailing Newline Violation: Files should have a single trailing newline. (trailing_newline)
/XcodeProjects/Prototypes/SwiftlintWeakDelegateIssue/SwiftlintWeakDelegateIssue/AppDelegate.swift:14:1: warning: Vertical Whitespace Violation: Limit vertical whitespace to a single empty line. Currently 3. (vertical_whitespace)
/XcodeProjects/Prototypes/SwiftlintWeakDelegateIssue/SwiftlintWeakDelegateIssue/AppDelegate.swift:34:1: warning: Vertical Whitespace Violation: Limit vertical whitespace to a single empty line. Currently 2. (vertical_whitespace)
/XcodeProjects/Prototypes/SwiftlintWeakDelegateIssue/SwiftlintWeakDelegateIssue/SceneDelegate.swift:15:1: warning: Line Length Violation: Line should be 120 characters or less: currently 127 characters (line_length)
/XcodeProjects/Prototypes/SwiftlintWeakDelegateIssue/SwiftlintWeakDelegateIssue/SceneDelegate.swift:16:1: warning: Line Length Violation: Line should be 120 characters or less: currently 122 characters (line_length)
/XcodeProjects/Prototypes/SwiftlintWeakDelegateIssue/SwiftlintWeakDelegateIssue/SceneDelegate.swift:18:1: warning: Line Length Violation: Line should be 120 characters or less: currently 147 characters (line_length)
/XcodeProjects/Prototypes/SwiftlintWeakDelegateIssue/SwiftlintWeakDelegateIssue/SceneDelegate.swift:26:1: warning: Line Length Violation: Line should be 120 characters or less: currently 140 characters (line_length)
/XcodeProjects/Prototypes/SwiftlintWeakDelegateIssue/SwiftlintWeakDelegateIssue/SceneDelegate.swift:52:1: warning: Trailing Newline Violation: Files should have a single trailing newline. (trailing_newline)
/XcodeProjects/Prototypes/SwiftlintWeakDelegateIssue/SwiftlintWeakDelegateIssue/SceneDelegate.swift:19:19: warning: Unused Optional Binding Violation: Prefer `!= nil` over `let _ =` (unused_optional_binding)
/XcodeProjects/Prototypes/SwiftlintWeakDelegateIssue/SwiftlintWeakDelegateIssue/SceneDelegate.swift:14:1: warning: Vertical Whitespace Violation: Limit vertical whitespace to a single empty line. Currently 2. (vertical_whitespace)
/XcodeProjects/Prototypes/SwiftlintWeakDelegateIssue/SwiftlintWeakDelegateIssue/SceneDelegate.swift:50:1: warning: Vertical Whitespace Violation: Limit vertical whitespace to a single empty line. Currently 2. (vertical_whitespace)
Done linting! Found 16 violations, 0 serious in 3 files.
```

### Environment

* SwiftLint version: 0.48.0
* Installation method used: either homebrew or Cocoapods
* Paste your configuration file: no configuration file

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
No
* Which Xcode version are you using (check `xcodebuild -version`)?
Xcode 13.3.1
Build version 13E500a
* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

```swift
protocol CoordinatorDelegate {
    func coordinatorDidFinish(_ coordinator: Coordinator)
}
final class Coordinator {
    var delegate: CoordinatorDelegate? // swiftlint should produce weak_delegate violation here
}
```

weak_delegate violation works fine on 0.45.1 version

**Comment by jpsim** [comment]
You'll need to explicitly enable it as it became disabled by default in 31872aaa3fa2674dc799755550f267a57e48fe2b (starting with [0.46.1](https://github.com/realm/SwiftLint/releases/tag/0.46.1)).
